kafka: add production docs (#101)
kudobuilder/frameworks: repository/kafka/README.md

![kudo-kafka](./docs/latest/resources/images/kudo-kafka.png)

# KUDO Kafka Operator

The KUDO Kafka operator creates, configures and manages [Apache Kafka](https://kafka.apache.org/) clusters running on Kubernetes.

### Overview

KUDO Kafka is a Kubernetes operator built on [KUDO](kudo.dev) to manage Apache Kafka in a scalable, repeatable, and standardized way over Kubernetes. Currently KUDO Kafka supports:

> - Secure Apache Kafka Clusters, support for **TLS  encryption** and **authentication**.

Reviewer: ```suggestion
- Securing the cluster in various ways: TLS encryption, Kerberos authentication, Kafka AuthZ
```

> - Metrics out of the box using **Prometheus**, and **Grafana** dashboard.

Reviewer: ```suggestion
- Prometheus metrics right out of the box with example Grafana dashboards
```

> - **Kerberos** support.

Reviewer: Probably not needed to emphasize these specific words like Kerberos, Prometheus, TLS encryption. IMO listing them is enough.

> - Graceful **rolling upgrades** for the cluster configuration and operator version.

Reviewer: ```suggestion
- Graceful rolling updates for any cluster configuration changes
- Graceful rolling upgrades when upgrading the operator version
```
Is this suggestion accurate? I just wanted to separate that rolling restarts can be both for updates and upgrades

Author: yeah right, those are the two main rolling restarts

To get more information around KUDO Kafka architecture please take a look on the [KUDO Kafka concepts](./docs/latest/concepts.md) document.

## Getting started

The latest stable version of Kafka operator is `1.0.0`

> For more details, please see the [docs](./docs/v1.0) folder.

Reviewer: ```suggestion
For more details, please see the [v1.0 docs](./docs/v1.0) folder.
```

> For the latest master branch you can check  [docs](./docs/latest) docs 

Reviewer: Let's remove this since we're pointing users to 1.0.0 docs right above

## Version Chart

| KUDO Kafka Version | Apache Kafka Version | Status |
| ------------------ | -------------------- | ------ |
| 0.1.2              | 2.2.1                | beta   |
| 0.2.0              | 2.3.0                | beta   |
| **1.0.0**          | **2.3.0**            | **GA** |

> | latest             | 2.3.0                | beta   |

Reviewer: We should fix this as soon as we merge this PR so that it doesn't give any window to confuse users

Reviewer: Might be worth putting the KUDO version too

Author: I added, as a minimum KUDO version

![kudo-kafka](./docs/latest/resources/images/kudo-kafka.png)

# KUDO Kafka Operator

The KUDO Kafka operator creates, configures and manages [Apache Kafka](https://kafka.apache.org/) clusters running on Kubernetes.

### Overview

KUDO Kafka is a Kubernetes operator built on [KUDO](kudo.dev) to manage Apache Kafka in a scalable, repeatable, and standardized way over Kubernetes. Currently KUDO Kafka supports:

- Securing the cluster in various ways: TLS encryption, Kerberos authentication, Kafka AuthZ
- Prometheus metrics right out of the box with example Grafana dashboards
- **Kerberos** support.
- Graceful rolling updates for any cluster configuration changes
- Graceful rolling upgrades when upgrading the operator version

To get more information around KUDO Kafka architecture please take a look on the [KUDO Kafka concepts](./docs/latest/concepts.md) document.

## Getting started

The latest stable version of Kafka operator is `1.0.0`
For more details, please see the [v1.0 docs](./docs/v1.0) folder.

> For the latest master branch you can check  [docs](./docs/latest) docs 

Reviewer 2: ```suggestion
For the latest master branch you can check the [latest docs](./docs/latest).
```

Author: going to remove this line as we agreed to just link to stable docs in `README.md`
